pkg/ccl/backupccl/restore_planning.go

```diff
@@ -158,6 +161,242 @@ func synthesizePGTempSchema(
+		// `system` database when looking for a conflict with system tables in the
+		// restoring cluster. This is because we do not create new descriptors for
+		// the backed up system tables but instead `DELETE` and `INSERT` rows into
+		// the restoring clusters' system tables.
+		for _, sysTableDesc := range systemTableDescs {
+			id := sysTableDesc.GetID()
+			if desc, ok := databasesByID[id]; ok {
+				if desc.GetID() != systemschema.SystemDB.GetID() {
+					systemTablesWithConflictingIDs[id] = sysTableDesc
+				}
+			} else if desc, ok := schemasByID[id]; ok {
+				if desc.GetParentID() != systemschema.SystemDB.GetID() {
+					systemTablesWithConflictingIDs[id] = sysTableDesc
+				}
+			} else if desc, ok := typesByID[id]; ok {
+				if desc.GetParentID() != systemschema.SystemDB.GetID() {
+					systemTablesWithConflictingIDs[id] = sysTableDesc
+				}
+			} else if desc, ok := tablesByID[id]; ok {
+				if desc.GetParentID() != systemschema.SystemDB.GetID() {
+					systemTablesWithConflictingIDs[id] = sysTableDesc
+				}
+			}
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return systemTablesWithConflictingIDs, nil
+}
+
+// moveConflictingSystemTables moves any system tables whose descriptor ID
+// conflicts with an ID in one of the provided maps to a new ID.
+//
+// TODO(ssd): We currently look for the list of conflicts in one transaction and
+// then move each table in its own transaction. I suppose this could be a
+// problem if there is some concurrent migration running.
+func moveConflictingSystemTables(
+	ctx context.Context,
+	p sql.PlanHookState,
+	databasesByID map[descpb.ID]*dbdesc.Mutable,
+	schemasByID map[descpb.ID]*schemadesc.Mutable,
+	tablesByID map[descpb.ID]*tabledesc.Mutable,
+	typesByID map[descpb.ID]*typedesc.Mutable,
+) error {
+	ctx, span := tracing.ChildSpan(ctx, "backupccl.moveConflictingSystemTables")
+	defer span.Finish()
+
+	systemTablesWithConflictingIDs, err := findConflictingSystemTables(ctx, p, databasesByID, schemasByID, tablesByID, typesByID)
+	if err != nil {
+		return errors.Wrapf(err, "failed to collect system table descriptor ID conflicts")
+	}
+
+	if len(systemTablesWithConflictingIDs) == 0 {
+		log.Infof(ctx, "no system table descriptor IDs conflict with IDs found in the backup")
+		return nil
+	}
+
+	var conflictPretty strings.Builder
+	for _, table := range systemTablesWithConflictingIDs {
+		fmt.Fprintf(&conflictPretty, " %s(%d)", table.GetName(), table.GetID())
+	}
+	log.Infof(ctx, "found %d system tables with descriptor IDs that conflict with IDs from the backup: %s",
+		len(systemTablesWithConflictingIDs),
+		conflictPretty.String(),
+	)
+
+	// We bump the DescIDSeqKey to an ID higher than anything in the backup to
+	// ensure that the tables we create below won't conflict with anything in the
+	// backup.
+	if err := bumpDescIDSequenceKey(ctx, p, databasesByID, schemasByID, tablesByID, typesByID); err != nil {
```

Comment:
I'm trying to piece together how this call interacts with the earlier call our caller made to bump it, before calling this function, if it was in the cluster restore case... after _that_ call we then inc it to get the tmp sys db ID, so what is this call going to bump it _to_?

Reply:
Discussed offline, because we take the max of the descriptor IDs in the backup, and the restoring clusters' sequence key we should be fine even if this method is called twice during restore planning.